DESIGN.md

```diff
@@ -0,0 +1,144 @@
+# libp2p testing story
+
+- Status: Draft
+
+---
+
+## Overview
+
+This document describes our process for testing interoperability & backward compatibility in libp2p.
+
+**Why care about this:**
+
+- Interoperability is a shared problem
+    - we don’t have a single blessed reference implementation that we use for conformance testing.
+    - No single maintainer (whether libp2p or ipdx) will succeed without everyone's involvement.
+- We want to share a Testing Story with the world that shows we care about quality & interop.
+- We want to encourage other implementations to join the testing party.
+
+**Context:**
+
+- We completed a “PING” interop test with Testground. It is running in go and rust-libp2p CIs.
+- It means we “proved” that we can write and run interop tests between versions AND implementation.
+
+# Libp2p Testing Matrix
+
+*What do we want to test next?*
+
+|                                   | go-libp2p | rust-libp2p | js-libp2p (node) | js-libp2p (browser) | jvm-libp2p | nim-libp2p |
+| ---                               | ---       | ---         | ---              | ---                 | ---        | ---        |
+| Simple PING [#35][issue-35]       | ✅        | ✅          | 🍎               | 🔥                  |            |            |
+| Circuit Relay                     |           |             |                  |                     |            |            |
+| WebTransport Transport            | 🔥        | 🔥          | 🔥               | 🔥                  | 🔥         | 🔥         |
```

Comment:
For webtransport, only Go is possible to test now (as only one that has the transport implemented).  JS is the only other implementation in-process on adding the transport.

```suggestion
| WebTransport Transport            | 🔥        |           | 🔥 (depends on https://github.com/libp2p/js-libp2p-webtransport/issues/1)               | 🔥 (depends on https://github.com/libp2p/js-libp2p-webtransport/issues/1)                  |          |          |
```